packages/world/src/WorldProxy.sol

```diff
@@ -44,6 +40,10 @@ contract WorldProxy is Proxy {
     StorageSlot.getAddressSlot(IMPLEMENTATION_SLOT).value = newImplementation;
 
     emit IERC1967.Upgraded(newImplementation);
+
+    IBaseWorld world = IBaseWorld(newImplementation);
+    emit IStoreEvents.HelloStore(world.storeVersion());
+    emit IWorldEvents.HelloWorld(world.worldVersion());
```

Comment:
although this isn't a place we need to really gas optimize, I kinda like what you had before 
```suggestion
    emit IStoreEvents.HelloStore(IBaseWorld(world).storeVersion());
    emit IWorldEvents.HelloWorld(IBaseWorld(world).worldVersion());
```

Reply:
Sure, I'm neither here nor there. Cool to try them both